webapp/channels/src/components/admin_console/custom_url_schemes_setting.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import * as Utils from 'utils/utils';
+import {t} from 'utils/i18n';
+
+import LocalizedInput from 'components/localized_input/localized_input';
+
+import Setting from './setting';
+
+const arrayToString = (arr: string[]) => {
+    return arr.join(',');
+};
+
+const stringToArray = (str: string) => {
+    return str.split(',').map((s) => s.trim()).filter(Boolean);
+};
+
+type Props = {
+    id: string;
+    value: string[];
+    onChange: (id: string, value: string[]) => void;
+    disabled?: boolean;
+    setByEnv: boolean;
+};
+
+const CustomURLSchemesSetting = ({
+    id,
+    value,
+    onChange,
+    disabled = false,
+    setByEnv,
+}: Props) => {
+    const [state, setState] = React.useState(arrayToString(value));
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const valueAsArray = stringToArray(e.target.value);
+
+        onChange(id, valueAsArray);
+
+        setState(e.target.value);
+    };
+
+    const label = Utils.localizeMessage('admin.customization.customUrlSchemes', 'Custom URL Schemes:');
```

Comment:
do we have any advantage of using this vs formatMessage?

Reply:
I'm not sure at all. I just copied over the existing code when changing it, didn't want to make too many changes that could bog this PR down. Let me look at changing it to `formatMessage` and see what happens.

Reply:
I asked over in community [here](https://community.mattermost.com/core/pl/mm5s3q8oni8kfp5xshbw18it3c). I'll update this once i get that back.